fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreIndexTest.java

```diff
@@ -2073,5 +2075,40 @@ public void formerIndexes() throws Exception {
             context.commit();
         }
     }
-    
+
+    @Test
+    public void boundaryPrimaryKeys() throws Exception {
+        final String indexName = "MySimpleRecord$num_value_unique";
+        try (FDBRecordContext context = openContext()) {
+            openSimpleRecordStore(context, TEST_SPLIT_HOOK);
+            recordStore.markIndexWriteOnly(indexName).get();
+            commit(context);
+        }
+
+        String bigOlString = Strings.repeat("x", SplitHelper.SPLIT_RECORD_SIZE + 2);
+        for (int i = -25; i < 25; i++) {
+            // Sparsify the primary keys so the records can be saved across boundaries.
+            saveAndSplitSimpleRecord(i * 39, bigOlString, i);
+        }
+
+        Index index = recordStore.getRecordMetaData().getIndex(indexName);
+        try (OnlineIndexer indexer = OnlineIndexer.newBuilder()
+                .setDatabase(fdb).setRecordStore(recordStore).setIndex(index)
+                .build()) {
+
+            List<Tuple> boundaryPrimaryKeys = recordStore.context.asyncToSync(FDBStoreTimer.Waits.WAIT_GET_BOUNDARY, indexer.buildEndpoints()
+                    .thenCompose(indexer::getBoundaryPrimaryKeysAsync
+                    ));
+
+            assertTrue(boundaryPrimaryKeys.size() > 2, "the test is meaning less if the records are not across boundaries");
```

Comment:
Out of curiosity, how many boundaries are there here?  Would it make sense to have a test that we run manually that are based upon a BUNCH of data?  Maybe not for this PR....I'm just thinking out loud.